src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -84,7 +83,7 @@ internal sealed class MeterProviderSdk : MeterProvider
             Func<Instrument, bool> shouldListenTo = instrument => false;
             if (meterSources.Any(s => s.Contains('*')))
             {
-                var regex = GetWildcardRegex(meterSources);
+                var regex = this.GetWildcardRegex(meterSources);
                 shouldListenTo = instrument => regex.IsMatch(instrument.Meter.Name);
             }
```

Comment:
Do we need a to a contains check for * and ? now?

Reply:
YES 👍

Reply:
I've fixed this and added test cases.
Not sure if it'll be better to extract the `s.Contains('*') || s.Contains('?')` logic into a common helper, looks like it is currently used in 3 places.

Reply:
> Not sure if it'll be better to extract the `s.Contains('*') || s.Contains('?')` logic into a common helper, looks like it is currently used in 3 places.

I think it'd be a decent minor improvement, so something like:

```csharp
internal static bool ContainsWildcard(string s)
{
    return s.Contains('*') || s.Contains('?');
}

...

if (meterSources.Any(ContainsWildcard))
```

but, I'm ok 👍 without if you think it reads better without the helper

Reply:
Maybe an extension method on String class? 😆

Reply:
That'd work too!

Reply:
I've updated the PR by extracting the wildcard logics to `WildcardHelper`.

I didn't use String extension method because it could be too noisy (e.g. if someone is using Intellisense).

Reply:
> I didn't use String extension method because it could be too noisy (e.g. if someone is using Intellisense).

Yea I think that's a good call. I was thinking that too, but thought maybe since it'd be internal it'd just affect us and not the whole 🌏